frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,625 @@
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([ROOT_COLLECTION, COLLECTION, SUBCOLLECTION]);
+
+          // dashboard item
+          expect(
+            screen.getByText(dashboardInPublicSubcollection.name),
+          ).toBeInTheDocument();
+        });
+
+        it("should show the root collection if the question is in a personal collection", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            mostRecentlyViewedDashboard: dashboardInPublicSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([ROOT_COLLECTION]);
+
+          // Showing personal collection means we're viewing the root collection
+          expect(
+            screen.getByRole("heading", { name: PERSONAL_COLLECTION.name }),
+          ).toBeInTheDocument();
+        });
+      });
+
+      describe("when last visited dashboard belongs to personal subcollections", () => {
+        const dashboardInPersonalSubcollection = createMockDashboard({
+          id: 5,
+          name: "Dashboard in personal subcollection",
+          collection: PERSONAL_SUBCOLLECTION,
+          collection_id: PERSONAL_SUBCOLLECTION.id as number,
+          model: "dashboard",
+        });
+
+        it("should show most recently visited dashboard if the question is in a public collection", async () => {
+          await setup({
+            mostRecentlyViewedDashboard: dashboardInPersonalSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([
+            ROOT_COLLECTION,
+            PERSONAL_COLLECTION,
+            PERSONAL_SUBCOLLECTION,
+          ]);
+
+          // dashboard item
+          expect(
+            await screen.findByText(dashboardInPersonalSubcollection.name),
+          ).toBeInTheDocument();
+        });
+
+        it("should show most recently visited dashboard if the question is in a personal collection", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            mostRecentlyViewedDashboard: dashboardInPersonalSubcollection,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([
+            ROOT_COLLECTION,
+            PERSONAL_COLLECTION,
+            PERSONAL_SUBCOLLECTION,
+          ]);
+
+          // dashboard item
+          expect(
+            await screen.findByText(dashboardInPersonalSubcollection.name),
+          ).toBeInTheDocument();
+        });
+      });
+    });
+
+    describe("when user didn't visit any dashboard during last 24hrs", () => {
+      it("should render root collection without preselection", async () => {
+        await setup();
+
+        // breadcrumbs show root collection only
+        expect(screen.getByTestId("item-picker-header")).toHaveTextContent(
+          ROOT_COLLECTION.name,
+        );
+      });
+
+      describe("question is in a public collection", () => {
+        describe('"Create a new dashboard" option', () => {
```

Comment:
test Query builder > Newly created question > #3, #4 from [the epic](https://github.com/metabase/metabase/issues/34733)